embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/DataSource.kt

```diff
@@ -51,20 +55,55 @@ internal interface DataSource<T> {
 }
 
 /**
- * A [DataSource] that adds either a [EmbraceSpanEvent] or [EmbraceSpanAttribute]
- * to the current session span.
+ * A [DataSource] that adds or alters a new span on the [SpansService]
  */
-internal typealias EventDataSource = DataSource<SessionSpanWriter>
+internal interface SpanDataSource : DataSource<TracingApi> {
```

Comment:
It seems to make more sense for this to get a `SpanService` instead and have all of its APIs be available? The idea is that these are internal to Embrace (for now), so we can grant them "private" access?